docs/Developers/Newcomers_Guide.rst

```diff
@@ -239,9 +239,27 @@ where ``myfork`` is the name of your fork, and ``fork_link`` is a link to your
 fork repository.
 
 .. note::
-   It is important that you do not make your changes on the master branch. To
-   start working on an issue, you first need to create a new branch where you
-   will work.
+   It is important that you **DO NOT** make your changes on the master branch
+   of your forked repository to avoid the following cases:
+
+   - If you make a rebase to synchronize your repository to the original,
+     every commit that is pushed to the remote master will be pulled in your
+     master branch. Then if you make a pull request to commit your changes to
+     the remote, the commits that got synced from the rebase will be
+     recommitted along with your work in the pull request.
+
+   - If your fork's master has been used in a pull request and you decide to
+     work on a different issue, you can either do that on the master branch,
```

Comment:
This isnt explained well.  There is only one branch named "master" .  A developer can not have two pull requests using the same branch name.